src/sentry/api/endpoints/project_rule_details.py

```diff
@@ -228,6 +229,15 @@ def put(self, request: Request, project, rule) -> Response:
         - Filters - help control noise by triggering an alert only if the issue matches the specified criteria.
         - Actions - specify what should happen when the trigger conditions are met and the filters match.
         """
+        rule_data_before = dict(rule.data)
+        if rule.environment_id:
+            rule_data_before["environment_id"] = rule.environment_id
+        if rule.owner:
+            rule_data_before["owner"] = rule.owner
+        rule_data_before["label"] = rule.label
+        rule_data_before["action_match"] = rule.data.get("action_match")
+        rule_data_before["filter_match"] = rule.data.get("filter_match")
```

Comment:
It seems like`dict(rule.data)` would already set these, is this being done so the values gets set to `None` if they're missing?

Reply:
lol no idk why I did this, probably because I added support for these last and didn't notice I already had it